liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/GenerateFeaturesMojo.java

```diff
@@ -277,11 +299,62 @@ private void addGenerationCommentToConfig(ServerConfigXmlDocument doc, File serv
+                // A RuntimeException means the currentFeatureSet contains conflicts.
+                // A FeatureConflictException means the binary files scanned conflict with each other or with
+                // the currentFeatureSet parameter.
+                Throwable scannerException = ite.getCause();
+                if (scannerException instanceof RuntimeException) {
+                    // The list of features from the app is passed in but it contains conflicts 
+                    String problemMessage = scannerException.getMessage();
+                    if (problemMessage == null || problemMessage.isEmpty()) {
+                        log.debug("RuntimeException from binary scanner without descriptive message", scannerException);
+                    } else {
+                        Set<String> conflicts = parseScannerMessage(problemMessage);
+                        Set<String> sampleFeatureList = null;
+                        try {
+                            sampleFeatureList = runBinaryScanner(null, getBinaryInputs(null, getClassesDirectories()));
+                        } catch (InvocationTargetException retryException) {
+                            // binary scanner should not return a RuntimeException since there is no list of app features passed in
+                            sampleFeatureList = new HashSet<String>();
+                            sampleFeatureList.add(BINARY_SCANNER_CONFLICT_MESSAGE4);
+                        }
+                        throw new RecommendationSetException(true, conflicts, sampleFeatureList);
+                    }
+                } else if (scannerException.getClass().getName().endsWith("FeatureConflictException")) {
+                    // The scanned files conflict with each other or with current features
+                    Set<String> conflicts = getConflicts(scannerException);
+                    Set<String> sampleFeatureList = null;
+                    if (currentFeatureSet != null) {
+                        try {
+                            sampleFeatureList = runBinaryScanner(null, getBinaryInputs(null, getClassesDirectories()));
+                        } catch (InvocationTargetException retryException) {
+                            Throwable scannerSecondException = retryException.getCause();
+                            if (scannerSecondException.getClass().getName().endsWith("FeatureConflictException")) {
+                                // Even after removing the server.xml feature list there are still conflicts in the binaries
+                                throw new NoRecommendationException(conflicts);
+                            } else {
+                                log.debug("Unexpected failure on retry call to binary scanner", scannerSecondException);
+                                log.debug("Passed directories to binary scanner:"+getClassesDirectories());
```

Comment:
In this case `sampleFeatureList` could be empty, then should we log this as an error or throw an error? I think right now it would continue on to throw a `RecommendationSetException`, which might deliver a confusing message to users.

Reply:
We're not sure what happened so I think the message4 response would deliver a satisfactory message.